compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -404,27 +404,45 @@ void RecordMinMax::profileDataWithRandomInputs(const std::string &mode, float mi
         throw std::runtime_error("Only support non-zero sized inputs");
 
       // TODO Support more input data types
-      assert(input_node->dtype() == loco::DataType::FLOAT32 ||
-             input_node->dtype() == loco::DataType::BOOL);
-
-      if (input_node->dtype() == DataType::FLOAT32)
+      switch (input_node->dtype())
       {
-        std::vector<float> input_data(num_elements);
+        case DataType::FLOAT32:
+        {
+          std::vector<float> input_data(num_elements);
 
-        // Write random data
-        for (auto &iter : input_data)
-          iter = static_cast<float>(dist(gen));
+          // Write random data
+          for (auto &iter : input_data)
+            iter = static_cast<float>(dist(gen));
 
-        // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
-        //       We can redcue the copy by directly writing data from file to interpreter inputs
-        _interpreter->writeInputTensor(input_node, input_data.data(),
-                                       input_data.size() * sizeof(float));
-      }
-      else if (input_node->dtype() == DataType::BOOL)
-      {
-        auto input_data = genRandomBoolData(gen, num_elements);
-        _interpreter->writeInputTensor(input_node, input_data.data(),
-                                       input_data.size() * sizeof(uint8_t));
+          // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
+          //       We can redcue the copy by directly writing data from file to interpreter inputs
+          _interpreter->writeInputTensor(input_node, input_data.data(),
+                                         input_data.size() * sizeof(float));
+          break;
+        }
+        case DataType::BOOL:
+        {
+          auto input_data = genRandomBoolData(gen, num_elements);
+          _interpreter->writeInputTensor(input_node, input_data.data(),
+                                         input_data.size() * sizeof(uint8_t));
+          break;
+        }
+        case DataType::S32:
+        {
+          auto input_data = genRandomIntData<int32_t>(gen, num_elements, -100, 100);
+          _interpreter->writeInputTensor(input_node, input_data.data(),
+                                         input_data.size() * sizeof(int32_t));
+          break;
+        }
+        case DataType::S64:
+        {
+          auto input_data = genRandomIntData<int64_t>(gen, num_elements, -100, 100);
+          _interpreter->writeInputTensor(input_node, input_data.data(),
+                                         input_data.size() * sizeof(int64_t));
+          break;
+        }
+        default:
+          throw std::runtime_error("Unsupported input type, yet!");
```

Comment:
For reviewer: note this PR introduces throw of `runtime_error()` in case of unsupported input datatype and removes `assert`, which checks datatypes (see above).